.github/workflows/minimum_dependency_checker.yml

```diff
@@ -0,0 +1,77 @@
+name: Minimum Dependency Checker
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'requirements.txt'
+      - 'test-requirements.txt'
+      - 'core-requirements.txt'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+      - name: Run min dep generator - test requirements
+        id: min_dep_gen_test
+        uses: alteryx/minimum-dependency-generator@v2
+        with:
+          requirements_paths: 'test-requirements.txt'
+      - name: Save min test requirements and run diff
+        id: check_min_test
+        continue-on-error: true
+        run: |
+          mkdir /tmp/dependencies_updated_artifacts
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > /tmp/minimum_test_requirements.txt
+          diff /tmp/minimum_test_requirements.txt evalml/tests/dependency_update_check/minimum_test_requirements.txt
+      - name: if min test requirements have changed, write output file
+        if: steps.check_min_test.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_test.outputs.min_reqs }}" > evalml/tests/dependency_update_check/minimum_test_requirements.txt
+      - name: Run min dep generator - requirements
+        id: min_dep_gen_reqs
+        uses: alteryx/minimum-dependency-generator@v2
+        with:
+          requirements_paths: 'requirements.txt'
+      - name: Save min requirements and run diff
+        id: check_min_reqs
+        continue-on-error: true
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > /tmp/minimum_requirements.txt
+          diff /tmp/minimum_requirements.txt evalml/tests/dependency_update_check/minimum_requirements.txt
+      - name: if min requirements have changed, write output file
+        if: steps.check_min_reqs.outcome != 'success'
+        run: |
+          printf "${{ steps.min_dep_gen_reqs.outputs.min_reqs }}" > evalml/tests/dependency_update_check/minimum_requirements.txt
+      - name: Run min dep generator - core requirements
+        id: min_dep_gen_core
+        uses: alteryx/minimum-dependency-generator@v2
+        with:
+          requirements_paths: 'core-requirements.txt'
+      - name: Save min core requirements and run diff
```

Comment:
There's a way in GH actions to define a "function" which could accept an arg like the requirements file name, and then execute the diff and file write, and then we could call that "function" 3x instead of copy-pasting the code, right? Not required for this PR lol just curious